sips/constant_qbft_timeout.md

```diff
@@ -39,31 +39,58 @@ Attester duties can be submitted with a delay of 31 slots,
 and therefore we can optimize validator performance by having a quick timeout in the first rounds,
 so the committee could have a "second chance" in case the leading operator is faulty.
 
+**Catch up property**   
+Timeouts need to be structured such that nodes in different rounds can "catch" one another and ultimately reach consensus, otherwise a liveness issue is created.
+
 **Quick Timeout Period**
 
 Smaller `T` helps validator performance in the first rounds, 
 but becomes redundant in later rounds when the duty was already finalized.
 
 As a mitigation, we use a quick timeout period where `T=2s`. 
-Once a threshold round was reached the timeout is extended to `T=2m`
+Once a threshold round was reached the timeout becomes larger `T=2m`
+
+**Stop For Catchup**  
+To enjoy both the benefits of a constant timeouts and the catch property we introduce a stop for catchup mechanism. 
+A node will progress through the rounds (via timeout) until it hits round r. At round r it will bump to r+1 but will not start a timeout clock.
+Instead, it will wait for 2f+1 round change messages for round r+1 **OR** f+1 round change messages for round higher than r+1.
```

Comment:
It shouldn't be considered. The idea is that we wait for all honest nodes to catch up. We only know they caught up if we have a quorum.

Reply:
@GalRogozinski 

So we have the following cases:
- `round == r+1` => need `2f+1` RCs (quorum)
- `round > r+1` => need `f+1` RCs (partial quorum)

Why partial quorum is enough for `round > r+1` and not enough for `round == r+1`?

Let's consider the following scenario:
- the committee size is `4` (`f == 1`)
- `round == r+1`
- all nodes (1-4) go down, then nodes 1-3 go up, and node 4 remains down

From node 1's perspective, it would eventually (after 2 min timeout) receive round changes for round `r+2` (`current_round + 1`) from nodes 2 and 3, and produce a round change itself, which means node 1 has a quorum of RCs, changes round to r+2 and continues. The same happens on nodes 2 and 3 and they try to reach the consensus again. Is that correct?

But if they are not able to reach the quorum again, after 2 min timeout they send round changes for `r+3`. As the current round is `r+2` which satisfies `round > r+1`, only partial quorum is enough for now. Why does it differ from the previous round?

Reply:
Note, that if all nodes go down then we reset to the beginning, unless you assume nodes 1-3 persisted their state in disk (or maybe you meant connection goes down?).

Changing to the next round on partial quorum is what we do in QBFT. Waiting for full quorum is the change.
Pretty much this SIP says:
1. Do normal QBFT with const timeouts
2. at `r` stop doing QBFT and do something else (wait for full quorum instead of partial quorum)
3. Resume normal QBFT with const timeouts.

Reply:
@nkryuchkov and partial quorum is enough because it ensures that at least one honest node has done a round change.
If one has changed it is best all others change to sync with it

Reply:
> maybe you meant connection goes down?

Yes. Sorry, I wasn't clear enough